File: C++/661. Image Smoother.cpp
class Solution {
public:
    vector<vector<int>> imageSmoother(vector<vector<int>>& M) {
        int dx[] = {0,1,-1,0,1,1,-1,-1};
        int dy[] = {1,0,0,-1,1,-1,-1,1};
        int n = M.size(), m = M[0].size();
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<m;j++)
            {
                int s1 = M[i][j], s2 = 1;
                for(int k=0;k<8;k++)
                {
                    int x = i + dx[k];
                    int y = j + dy[k];
                    if(x<0 || x>=n || y<0 || y>=m)
                        continue;
                    s1 += (M[x][y] & 0xFF);
                    s2 += 1;
                }
                M[i][j] |= ((s1/s2) << 8);
            }
        }
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<m;j++)
                M[i][j] >>= 8;
        }
        return M;
    }
};

File: Python/661. Image Smoother.py
class Solution:
    def imageSmoother(self, M):
        """
        :type M: List[List[int]]
        :rtype: List[List[int]]
        """
        ans = copy.deepcopy(M)
        dx = [1,0,0,-1,1,1,-1,-1]
        dy = [0,-1,1,0,-1,1,-1,1]
        n, m = len(M), len(M[0])
        for i in range(n):
            for j in range(m):
                s1, s2 = M[i][j], 1
                for k in range(8):
                    x = i + dx[k]
                    y = j + dy[k]
                    if x < 0 or x >= n:
                        continue
                    if y < 0 or y >= m:
                        continue
                    s1 += M[x][y]
                    s2 += 1
                ans[i][j] = int(s1/s2)
        return ans